### PR TITLE
[CL-305] Allow margin to be disabled for elements at the bottom of a page

### PR DESCRIPTION
**Pull request by vleague2**
## 🎟️ Tracking

[CL-305](https://bitwarden.atlassian.net/browse/CL-305)

## 📔 Objective

This PR adds the ability to manually disable a `bit-section`'s bottom margin, primarily for use on sections that are the last element on the page and should not have extra bottom margin. This is a pattern used on other components with built-in margin.

This PR implements the `disableMargin` input on the extension vault page, which is where the extra margin issue was reported. It also adds a `section` to the `popup-layout` stories to provide a more realistic story for the extension layout.

## 📸 Screenshots

Check Storybook stories for section and popup layout.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[CL-305]: https://bitwarden.atlassian.net/browse/CL-305?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/browser/src/vault/popup/components/vault-v2/vault-list-items-container/vault-list-items-container.component.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<bit-section *ngIf="ciphers?.length > 0 || description">
+<bit-section *ngIf="ciphers?.length > 0 || description" disableMargin>
```
Comment (shane-melton) [inline apps/browser/src/vault/popup/components/vault-v2/vault-list-items-container/vault-list-items-container.component.html:1]: ⚠️ This container is used a few times in the `vault-v2` component, one for each list section, with some needing the bottom margin to separate the section lists (at least, as far as I'm aware). 

If so, we will likely want to make this an input to the list container to pass through and then set the attribute on the last `<vault-list-items-container>` in the `vault-v2` template.

**Review by shane-melton — CHANGES_REQUESTED** [review]

File: apps/browser/src/vault/popup/components/vault-v2/vault-list-items-container/vault-list-items-container.component.html (hunk 1 above)
Comment (vleague2) [reply apps/browser/src/vault/popup/components/vault-v2/vault-list-items-container/vault-list-items-container.component.html:1]: Oh man good catch 🤦‍♀️

**Review by shane-melton — DISMISSED** [review]
Looks great, thank you!

File: libs/components/src/section/section.component.ts [hunk 2]
```diff
@@ -1,14 +1,25 @@
...
-import { Component } from "@angular/core";
+import { Component, Input } from "@angular/core";
 
 @Component({
   selector: "bit-section",
   standalone: true,
   imports: [CommonModule],
   template: `
-    <section class="tw-mb-6 md:tw-mb-12">
+    <section [ngClass]="sectionClass">
```
Comment (willmartian) [inline libs/components/src/section/section.component.ts:10]: Thoughts on leaving the Tailwind classes in the template? Imo it is easier to parse. 
```
[ngClass]="{
  'tw-mb-6 md:tw-mb-12': !disableMargin
}"
```

**Review by willmartian — APPROVED** [review]
One non-blocking suggestion:

File: libs/components/src/section/section.component.ts (hunk 2 above)
Comment (vleague2) [reply libs/components/src/section/section.component.ts:10]: Agreed, idk what I was thinking 😂

Comment (willmartian) [reply libs/components/src/section/section.component.ts]: Same, all the time

**Review by willmartian — APPROVED** [review]
:shipit:

**Review by shane-melton — APPROVED** [review]

**Review by willmartian — APPROVED** [review]

**Review by shane-melton — APPROVED** [review]

**Pull request merged**